src/menu_base.cpp

```diff
@@ -1603,9 +1606,12 @@ void MenuBase::handleStandardInput(const std::string &character_buffer)
             return;
         }
 
+        // Extra test, if the menu changed, then don't clear input!
+        std::string tmp_menu = m_current_menu;
+
         // Process incoming String from Menu Input up to ENTER.
         // If no commands were processed, erase all prompt text
-        if (!processMenuOptions(key))
+        if (!processMenuOptions(key) && tmp_menu == m_current_menu)
```

Comment:
Fix for Menu Prompts here, make sure when the menu changes, we no longer try to clear displayed input text, this would end up overwrite or backspace over the menu prompt.  Now fixed.